flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/context/SessionContext.java

```diff
@@ -111,7 +127,8 @@ public void reset() {
         // If rebuild a new Configuration, it loses control of the SessionState if users wants to
         // modify the configuration
         resetSessionConfigurationToDefault(defaultContext.getFlinkConfig());
-        this.executionContext = new ExecutionContext(executionContext);
+        buildClassLoaderAndUpdateDependencies(defaultContext.getDependencies());
```

Comment:
Is it on purpose that `RESET` will also reset added jars? IIRC, `RESET` only reset configurations.